lib/postgrex_wal/messages/util.ex

```diff
@@ -0,0 +1,108 @@
+defmodule PostgrexWal.Messages.Util do
+  @moduledoc false
+  alias PostgrexWal.Message
+
+  @spec decode_lsn(lsn :: integer) :: String.t()
+  defdelegate decode_lsn(lsn), to: __MODULE__.LSN
+
+  @pg_epoch ~U[2000-01-01 00:00:00.000000Z]
+  @spec decode_timestamp(microsecond_offset :: integer) :: DateTime.t()
+  def decode_timestamp(microsecond_offset) when is_integer(microsecond_offset) do
+    DateTime.add(@pg_epoch, microsecond_offset, :microsecond)
+  end
+
+  @spec binary_split(binary, integer, binary) :: list(binary)
+  def binary_split(binary, parts \\ 2, delimeter \\ <<0>>) do
+    String.split(binary, delimeter, parts: parts)
+  end
+
+  @spec decode_tuple_data!(tuple_data :: binary) :: [Message.tuple_data()]
```

Comment:
这里别加 `!` 了，其他的 function 都没加
```suggestion
  @spec decode_tuple_data(tuple_data :: binary) :: [Message.tuple_data()]
```

Reply:
需要二种不同的函数。返回值也不同。

```elixir
@spec decode_tuple_data!(tuple_data :: binary) :: [Message.tuple_data()]
  def decode_tuple_data!(tuple_data) do
    {<<>>, decoded_tuple_data} = decode_tuple_data(tuple_data)  # 不匹配会抛错
    decoded_tuple_data
  end

@spec decode_tuple_data(tuple_data :: binary) :: {binary, [Message.tuple_data()]}
```

Reply:
另一种，就是换个函数名了。

`decode_tuple_data!/1` 换成 `get_tuple_data/1`